angular/src/app/shared/util.ts

```diff
@@ -10,3 +10,7 @@ export function objectToArray(dto) {
 
   return bar;
 }
+
+export function flattenArray(arr) {
```

Comment:
+1 for extracting

There is already one flatten function in `data-list.component.ts`:
```
  private static flatten<T>(arrayOfArrays: T[][]): T[] {
    return arrayOfArrays.reduce((accum, cur) => accum.concat(...cur), []);
  }
```

1. Please either use that signature or add types to your `flattenArray`, in that case it is enough to call it just `flatten` the "suffix" is implicit from type
2. Remove it from data-list and import from your util file